### ISSUE TITLE
Own Implementation of getShippingTaxClass

### ISSUE THREAD

**Issue by sprankhub** [PAYONE-GmbH/magento-1#64]
In https://github.com/PAYONE-GmbH/magento-1/blob/master/app/code/community/Payone/Core/Helper/Config.php, you have a method `getShippingTaxClassId`, which simply returns the store config from `Mage_Tax_Model_Config::CONFIG_XML_PATH_SHIPPING_TAX_CLASS`. Exactly the same method is available in the Magento core in `Mage_Tax_Model_Config::getShippingTaxClass`. Is there any reason why there is a custom implementation of it? I also checked older Magento versions, but this method even exists in [Magento 1.4](https://github.com/OpenMage/magento-mirror/blob/magento-1.4/app/code/core/Mage/Tax/Model/Config.php#L252). If there is no reason, I would like to remove the custom implementation and use the one from the Magento core. This makes customisations of the shipping tax class much easier.

Background: MageSetup rewrites [`Mage_Tax_Model_Config::getShippingTaxClass`](https://github.com/firegento/firegento-magesetup/blob/master/src/app/code/community/FireGento/MageSetup/Model/Tax/Config.php) and implements a custom shipping tax class algorithm. I fear that there are issues if PAYONE uses another shipping tax class.

**Comment by fjbender** [comment]
I don't see any reason why we would be doing that. It's only ever used in `Payone_Core_Model_Mapper_ApiRequest_Payment_Abstract` and we could use the `Mage_Tax_Model_Config` there, I suppose.

@FatchipRobert any objections?

**Comment by sprankhub** [comment]
Cool :)

On a related note, could you tell me when `Payone_Core_Model_Mapper_ApiRequest_Payment_Abstract` is used exactly? Thanks!

**Comment by fjbender** [comment]
```
florian@charon:~/Dev/magento-1 (master u=)$ grep -ir Payone_Core_Model_Mapper_ApiRequest_Payment_Abstract .
./app/code/community/Payone/Core/Model/Mapper/ApiRequest/Payment/Authorize/Abstract.php:    extends Payone_Core_Model_Mapper_ApiRequest_Payment_Abstract
./app/code/community/Payone/Core/Model/Mapper/ApiRequest/Payment/Abstract.php:abstract class Payone_Core_Model_Mapper_ApiRequest_Payment_Abstract
./app/code/community/Payone/Core/Model/Mapper/ApiRequest/Payment/Debit.php:    extends Payone_Core_Model_Mapper_ApiRequest_Payment_Abstract
./app/code/community/Payone/Core/Model/Mapper/ApiRequest/Payment/Capture.php:    extends Payone_Core_Model_Mapper_ApiRequest_Payment_Abstract
./app/code/community/Payone/Core/Model/Mapper/ApiRequest/Payment/Genericpayment.php:    extends Payone_Core_Model_Mapper_ApiRequest_Payment_Abstract
```

**Comment by sprankhub** [comment]
Hence, the shipping tax class is transmitted to PAYONE? Then the wrong shipping tax class is currently transmitted. Is this a severe issue? Not sure how the shipping tax class is used on the PAYONE side...

**Comment by fjbender** [comment]
Payone doesn't really need the tax class, only the tax rate will be transmitted:

```php
        $shippingTaxClass = $this->helperConfig()->getShippingTaxClassId($storeId);
        if ($shippingTaxClass) {
            $request->setProductClassId($shippingTaxClass);
            return $taxCalculationModel->getRate($request);
        }
        else {
            return 0.0;
        }
```

It's used for instance when Payone generates a PDF invoice.

**Comment by fjbender** [comment]
@sprankhub Can you take a look at #84 and see if this fits your needs?

Cheers
Florian

**Comment by sprankhub** [comment]
Looks good :+1: Thanks!

**Comment by fjbender** [comment]
Fixed with #84
